### ISSUE TITLE
The `arrow` package will be removed from CRAN

### ISSUE THREAD

**Issue by eitsupi** [pola-rs/r-polars#779]
See apache/arrow#39806.

We may have to update the CI.

**Comment by etiennebacher** [comment]
We must ensure that 

- [ ] we use `skip_if_not_installed()` in tests
- [ ] we use `requireNamespace()` in examples
- [ ] we check it's installed before using it in the code

**Comment by eitsupi** [comment]
I don't think there was any place where the functions of the arrow package are currently used except in the S3 methods of the functions of the arrow package.

**Comment by eitsupi** [comment]
#785 fixed tests, but there is actually a code block in the `polars` vignette that also uses `arrow`, so we need to fix that as well.

Also, Taskfile and other files need to be updated.

As for CI, it may be a hassle to set up to install Linux binaries from p3m, so in some cases it may be better not to set it up.

**Comment by etiennebacher** [comment]
> Also, Taskfile and other files need to be updated.

I don't think we need to update it since we use `require()` to load arrow and that only warns if the package is absent.
